avocado/core/task/statemachine.py

```diff
@@ -288,13 +256,25 @@ async def monitor(self):
                            'id': str(runtime_task.task.identifier),
                            'job_id': runtime_task.task.job_id}
                 self._state_machine._status_repo.process_message(message)
+        # from here, this `task` ran, so, let's check
+        # the its latest data in the status repo
+        latest_task_data = \
+            self._state_machine._status_repo.get_latest_task_data(
+                str(runtime_task.task.identifier)) or {}
+        # maybe, the results are not available yet
+        while latest_task_data.get("result", None) is None:
+            await asyncio.sleep(0.1)
+            latest_task_data = \
+                self._state_machine._status_repo.get_latest_task_data(
+                    str(runtime_task.task.identifier)) or {}
+        runtime_task.result = latest_task_data['result']
         result_stats = set(key.upper()for key in
                            self._state_machine._status_repo.result_stats.keys())
         if self._failfast and not result_stats.isdisjoint(STATUSES_NOT_OK):
             await self._state_machine.abort("FAILFAST is enabled")
             raise TestFailFast("Interrupting job (failfast).")
 
-        await self._state_machine.finish_task(runtime_task)
+        await self._state_machine.finish_task(runtime_task, "FINISHED")
```

Comment:
Maybe we can change the log message here. Because the important part of this change is, that the `finish_task` will set the `runtime_task` status to `FINISHED` and the status is used for detecting all finished tasks. in [here](https://github.com/avocado-framework/avocado/pull/5412/files#diff-b2cc6571a0856868fe60a323f79a542765b0fa23da3d10876423e3d6de73e7fcR68).

Reply:
> Maybe we can change the log message here. Because the important part of this change is, that the `finish_task` will set the `runtime_task` status to `FINISHED` and the status is used for detecting all finished tasks in [here](https://github.com/avocado-framework/avocado/pull/5412/files#diff-b2cc6571a0856868fe60a323f79a542765b0fa23da3d10876423e3d6de73e7fcR68).

I'm fine with that, and I overlooked the function of the status.  We can have the message something like:

```
Task "foo-bar" finished with status: FINISHED
```

That seems fine to me.